### PR TITLE
release-checklist.md: small improvements and updates

### PR DESCRIPTION
**Pull request by prestist**
Going through the steps for releasing ignition v2.14.0  https://github.com/coreos/ignition/issues/1353

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ Fedora packaging:
 
 GitHub release:
  - [ ] Wait until the Bodhi update shows "Signed :heavy_check_mark:" in the Metadata box.
- - [ ] [File a releng ticket](https://pagure.io/releng/new_issue) based on [prior signing tickets](https://pagure.io/releng/issue/10178).
+ - [ ] [File a releng ticket](https://pagure.io/releng/new_issue) based on [prior signing tickets](https://pagure.io/releng/issue/10799).
    - [ ] Use the updated script from https://github.com/coreos/ignition/issues/1311#issuecomment-1024662891, then PR this checklist to update the example ticket link.
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/release-checklist.md:45]: This line was temporary and should be dropped.

Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/release-checklist.md:45]: This line was temporary and should be dropped.

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 2]
```diff
@@ -20,19 +20,19 @@ Fedora packaging:
...
    git merge --ff-only rawhide
-   git push origin f35
+   git push origin f36
    ```
  - [ ] On each of those branches run `fedpkg build`
  - [ ] Once the builds have finished, submit them to [bodhi](https://bodhi.fedoraproject.org/updates/new), filling in:
    - `ignition` for `Packages`
    - Selecting the build(s) that just completed, except for the rawhide one (which gets submitted automatically)
-   - Writing brief release notes like "New upstream release; see release notes at `link to GitHub release`"
+   - Writing brief release notes like "New upstream release; see release notes at `link to NEWS on GH tag`"
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/release-checklist.md:35]: `NEWS.md` once #1380 lands.

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 3]
```diff
@@ -20,19 +20,19 @@ Fedora packaging:
...
    git checkout rawhide
    git pull --ff-only
-   git checkout f35
+   git checkout f36
    git merge --ff-only rawhide
-   git push origin f35
+   git push origin f36
    ```
- - [ ] On each of those branches run `fedpkg build`
+ - [ ] On each of those branches run `fedpkg build` including rawhide.
```
Comment (bgilbert) [inline .github/ISSUE_TEMPLATE/release-checklist.md:31]: Did you intend to have this in https://github.com/coreos/butane/pull/349 also?

Comment (prestist) [reply .github/ISSUE_TEMPLATE/release-checklist.md:31]: Yeah, I should add it there as well.

Comment (prestist) [reply .github/ISSUE_TEMPLATE/release-checklist.md:31]: done

**Review by bgilbert — APPROVED** [review]
Thanks!

**Pull request merged**